src/commands/postDeploy/checkLinuxWebAppDownDetector.ts

```diff
@@ -0,0 +1,176 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { WebResource } from "ms-rest";
+import { CancellationTokenSource } from "vscode";
+import { callWithTelemetryAndErrorHandling, IActionContext, openInPortal, UserCancelledError } from "vscode-azureextensionui";
+import { KuduClient } from "vscode-azurekudu";
+import { DeployResult } from "vscode-azurekudu/lib/models";
+import { SiteTreeItem } from "../../explorer/SiteTreeItem";
+import { ext } from '../../extensionVariables';
+import { delay } from "../../utils/delay";
+import { requestUtils } from '../../utils/requestUtils';
+import { getGlobalSetting } from '../../vsCodeConfig/settings';
+import { findTableByColumnName, findTableByRowValue, getValuesByColumnName } from "./parseDetectorResponse";
+
+const linuxContainerStartFailureId: string = 'LinuxContainerStartFailure';
+
+export enum ColumnName {
+    status = "Status",
+    message = "Message",
+    name = "Data.Name",
+    value = "Data.Value",
+    expanded = "Expanded",
+    solutions = "Solutions",
+    time = "Time",
+    instance = "Instance",
+    facility = "Facility",
+    failureCount = "FailureCount"
+}
+
+export async function checkLinuxWebAppDownDetector(correlationId: string, node: SiteTreeItem, tokenSource: CancellationTokenSource): Promise<void> {
+    return await callWithTelemetryAndErrorHandling('appService.linuxWebAppDownDetector', async (context: IActionContext): Promise<void> => {
+        context.errorHandling.suppressDisplay = true;
+        context.telemetry.properties.correlationId = correlationId;
+
+        const kuduClient: KuduClient = await node.root.client.getKuduClient();
+        const deployment: DeployResult = await kuduClient.deployment.getResult('latest');
+        if (!deployment.startTime) {
+            // if there's no deployment detected, nothing can be done
+            context.telemetry.properties.cancelStep = 'noDeployResult';
+            return;
+        }
+
+        const deployResultTime: Date = new Date(deployment.startTime);
+        const enableDetectorsSetting: string = 'enableDetectors';
+        const showOutput: boolean | undefined = getGlobalSetting<boolean>(enableDetectorsSetting);
```

Comment:
You should almost always use `getWorkspaceSetting`. There's no reason to ignore the value if it's set at the workspace level and `getWorkspaceSetting` will defer to the global setting if the workspace setting isn't defined

Reply:
ping on this one